proposal/custom-url-resolution.md

```diff
@@ -0,0 +1,117 @@
+
+If a certain resource cannot be remapped the url rewrite plugin will return `null`, if all rewrite plugins return `null` it should fallback to the original value of the url reference.
+
+### JavaScript API
+
+At the core of remapping the url imports is the JavaScript API which allows users to return a new url reference based on the filepath or content.
+
+The first parameter this rewriteUrl function gets passed in is the importerResult, which is an object containing an optional `file` and `content` field.
+
+- The `file` field is the location on disk of the given url resource, this can be used to create relative mappings or to simply read the file.
+- The `content` field is a `Blob` which contains the content of the url resource, in this case the url rewriter should not try to read the file manually and use this instead as this is what the importer returned.
+
+This function can be used both asynchronously and synchronously, in case of it being asynchronous it should call the done callback, otherwise it should directly return the url synchronously (which is a string or null).
+
+The done callback has two parameters, `error` and `url`:
+
+- The `error` parameter can be `null` or an `Error` object, this is used when an error occurred when trying to rewrite the url.
+- The `url` parameter can be `null` or a `string`, this should be the new url as it can be found in the browser.
+
+Asynchronous example:
+
+```TypeScript
+// Rewrite url plugin/function
+const rewriteUrlPlugin = async (importerResult: { file?: string, content?: Blob }, done: (error: Error | null, url: string | null) => void): void => {
+  if (file && !content) {
+    content = await fs.readFile(file);
+  }
+
+  done(null, content ? `data:${base64(content)))}` : null);
+}
+
+let sassOptions = {
+  // This can be an array or a single function
+  // depending on whether it's a pipeline of rewrite plugins or just a single plugin
+  rewriteUrl: [rewriteUrlPlugin]
+}
+```
+
+Synchronous example:
+
+```TypeScript
+let rootFile = '/example/test.scss';
+
+let sassOptions = {
+  // Could also be an array as shown in the async example
+  rewriteUrl: (importerResult: { file?: string, content?: Blob }): string | null => {
+    if (!importerResult.file) {
+      return null;
+    }
+
+    return path.relative(path.dirname(rootFile), file);
+  }
+}
+```
+
+### Edge cases
+
+#### Variables in the url reference
+
+Variables can be used in a url reference, in this case it should get remapped based on the value that gets created after the variables have been applied to ensure we are able to remap the url as expected.
+
+Examples:
+
+```Scss
+url("#{$asset-path}/image.png");
+```
+
+```Scss
+url("./folder/#{$some-var}");
+```
+
+## Syntax
+
+This proposal applies to the `url(...)` function, both with and without the quotation marks. This ensures this applies to the entire css specification for `url(...)`.
+
+This proposal does not introduce any new Syntax.
```

Comment:
Since this doesn't affect the Sass's parsing phase, you don't need to include a Syntax section. However, you should include a Semantics section that specifies exactly what the new process of resolving a URL is.